varinfo/umm_var.py

```diff
@@ -338,3 +341,53 @@ def get_json_serializable_value(input_value: Any) -> Any:
         output_value = input_value
 
     return output_value
+
+
+def publish_umm_var(collection_id: str,
+                    umm_var_dict: Dict,
+                    token: str,
+                    cmr_env: CMR_UAT) -> str:
+    """" Publish a single UMM-Var entry to CMR given:
+        * collection_id: a collection's concept_id
+        * umm_var_dict: a dictionary of a single UMM-Var entry for a collection
+        * token: Earthdata Login (EDL) bearer_token
+        * cmr_env/mode: CMR environments (OPS, UAT, and SIT)
+    For a successful publication response all of these fields must be entered
+
+    """
+    # Required UMM-Var headers for ingestiong variable entries
+    headers_umm_var = {
+        'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
+        + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
+        'Authorization': f'Bearer {token}',
```

Comment:
Honest question (maybe for Ryan): For publication, aren't we aiming to use a LaunchPad token, instead of an EDL one? I'm not as familiar with those, but I have a sneaky suspicion they aren't `Bearer` tokens. I wonder if maybe the best thing to do (here and in the search functionality) is to assume someone has passed us the full `Authorization` header, and then just pass that through as-is, rather than rebuilding the structure to (or not to) include the `Bearer` bit?

Reply:
Thanks! Do you or @flamingbear have an example of this? I am not too familiar with LaunchPad tokens either.

Reply:
I don't have an example, but it sounds like @owenlittlejohns is saying, rather than pass `token` as a param to publish_umm_var, just receive the whole auth header header as input. and add that.  

Something like 
```
def publish_umm_var(collection_id: str,
                    umm_var_dict: Dict,
                    auth_header: str,
                    cmr_env: CMR_UAT) -> str:
```

And then 

``` python
    headers_umm_var = {
        'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
        + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
        'Authorization': auth_header,
        'Accept': 'application/json'}
```

But that is just me guessing from what Owen typed and not knowing about the different types of tokens.

Reply:
Thank you! That makes sense!